python/intro/README.md

```diff
@@ -0,0 +1,405 @@
+correctly or not. Function testing is not covered in this tutorial.
+
+
+## Running Your First Function
+With the `pythonfn` directory containing `func.py` and `func.yaml` you've
+got everything you need to run the function.  So let's run it and
+observe the output.  Note that the first time you build a
+function of a particular language it takes longer as Fn downloads
+the necessary Docker images.
+
+![user input](images/userinput.png)
+>```sh
+> fn run
+>```
+
+```yaml
+Building image fndemouser/pythonfn:0.0.1
+{"message":"Hello World"}
+```
+
+The last line of output is `{"message":"Hello World"}` since no input was passed
+to the function.
+
+If you ever want more details on what a given fn command is doing behind the
+scenes you can add the `--verbose` switch.  Let's rerun with verbose output
+enabled.
+
+![user input](images/userinput.png)
+>```sh
+> fn --verbose run
+>```
+
+```yaml
+Building image pythonfn:0.0.1
+Sending build context to Docker daemon  6.144kB
+Step 1/6 : FROM python:3.6
+3.6: Pulling from library/python
+f2b6b4884fc8: Pull complete
+4fb899b4df21: Pull complete
+74eaa8be7221: Pull complete
+2d6e98fe4040: Pull complete
+414666f7554d: Pull complete
+135a494fed80: Pull complete
+6ca3f38fdd4d: Pull complete
+4de6fcaa1241: Pull complete
+Digest: sha256:f76c4384f42bbb9065aff0a444c2d61f7ba3d9d44873976b4179b4e34280c6bb
+Status: Downloaded newer image for python:3.6
+ ---> 07d72c0beb99
+Step 2/6 : WORKDIR /function
+Removing intermediate container 9351c142b904
+ ---> 3ce33feb8e51
+Step 3/6 : ADD requirements.txt /function/
+ ---> f41ea90c9672
+Step 4/6 : RUN pip3 install -r requirements.txt
+ ---> Running in b20ba0ee4603
+Collecting fdk (from -r requirements.txt (line 1))
+  Downloading fdk-0.0.25-py2.py3-none-any.whl
+Collecting iso8601==0.1.12 (from fdk->-r requirements.txt (line 1))
+  Downloading iso8601-0.1.12-py2.py3-none-any.whl
+... # Truncated for brevity
+Step 5/6 : ADD . /function/
+ ---> 392b18c683d0
+Step 6/6 : ENTRYPOINT ["python3", "func.py"]
+ ---> Running in 29f238ca5b19
+Removing intermediate container 29f238ca5b19
+ ---> 2e4c259d3723
+Successfully built 2e4c259d3723
+Successfully tagged pythonfn:0.0.1
+{"message":"Hello World"}
+```
+
+You can also pass data to the run command. For example:
+
+![user input](images/userinput.png)
+>```sh
+> echo -n '{"name":"Bob"}' | fn run
+>```
+
+```yaml
+Building image fndemouser/pythonfn:0.0.1
+{"message":"Hello Bob"}
+```
+
+The JSON data was parsed and since `name` was set to "Bob", that value is passed
+in the output.
+
+### Understanding fn run
+If you have used Docker before the output of `fn --verbose run` should look
+familiar--it looks like the output you see when running `docker build`
+with a Dockerfile.  Of course this is exactly what's happening!  When
+you run a function like this Fn is dynamically generating a Dockerfile
+for your function, building a container, and then running it.
+
+> __NOTE__: Fn is actually using two images.  The first contains
```

Comment:
you need to remove this, since python is not a compiled language, so it doesn't use multi-stage builds.

Reply:
Good to know. Removed.